openapi/generator_test.go

```diff
@@ -190,13 +136,26 @@ func TestSchemaFromMapWithUnsupportedKeys(t *testing.T) {
 	assert.NotEmpty(t, g.Errors()[0].Error())
 }
 
+// TestSchemaFromDuplicatedType tests that a
+// schema cannot be created with 2 types with
+// the same name
+func TestSchemaFromDuplicatedType(t *testing.T) {
+	g := gen(t)
+
+	schema := g.newSchemaFromType(rt(new(extraTypes.D)))
+	assert.NotNil(t, schema)
+	assert.Len(t, g.Errors(), 1)
+	assert.Implements(t, (*error)(nil), g.Errors()[0])
```

Comment:
Could you also test that the error is a `TypeError` containing the message `encountered duplicated type`, where the `error.T` is effectively `extratypes.D` ?

Reply:
Done 👍